### PR TITLE
Fixing fatal error when dispatch returns WP_Error

### PR DESCRIPTION
**Pull request by davefx**
Replicating treatment of potential WP_Error that is done in the serve_request() method to avoid fatal errors while processing WP_Error as if it were a WP_REST_Response.

Trac ticket: [#56566](https://core.trac.wordpress.org/ticket/56566).

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/rest-api/class-wp-rest-server.php [hunk 1]
```diff
@@ -727,8 +727,16 @@ protected function embed_links( $data, $embed = true ) {
 
 					$response = $this->dispatch( $request );
 
+					// Normalize to either WP_Error or WP_REST_Response...
+					$result = rest_ensure_response( $result );
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api/class-wp-rest-server.php]:
```suggestion
					$response = rest_ensure_response( $response );
```

Comment (davefx) [reply src/wp-includes/rest-api/class-wp-rest-server.php]: Fixed! Sorry for the error!

Comment (TimothyBJacobs) [reply src/wp-includes/rest-api/class-wp-rest-server.php]: No worries!

**Comment by davefx** [comment]
Closing PR, as it's already included in PR #3270.

**Pull request closed without merging**